docs/kola: adding instructions to using and testing kola (#2738)
coreos/coreos-assembler: docs/kola-guide.md

 3. `cosa kola run <test-pattern>` (This will run a specific test and check to see if a test passes)
 4. `cosa kola run ext.config*` (This will run all the tests that start with `ext.config`)

### More information on tests

After you run the kola test, you can find more information in tmp/kola/<test-name> about the test that just ran, as the following file logs . They will help you to debug the problem and will certainly give you hints along the way

1. `journal.txt`
2. `console.txt`
3. `ignition.json`
4. `gunzip journal-raw.txt.gz`

### Extended artifacts

1. Extended artifacts need additional forms of testing (You can pass the ignition and the path to the artifact you want to test)
2. `cosa kola run -h` (this allows you to see the commands yourself and what syntax is needed)
3. `cosa buildextend-"name_of_artifact` (An example of building an artifact before testing it)
4. `kola run -p <platform>` Is the most generic way of testing extended artifacts, this is mostly useful for the cloud platforms
5. For running the likes of metal/metal4k artifacts there's not much difference than running `kola run` from the coreos-assembler

> 6. `cd builds/latest/` (This will show your latest build information)

Reviewer: I'd use `cosa list` here; it will show you the most recent CoreOS builds that have been made and the artifacts created:

```
[coreos-assembler]$ cosa list
35.20220302.dev.0
   Timestamp: 2022-03-02T14:26:11Z (5 days, 1:59:15 ago)
   Artifacts: ostree qemu
      Config: pr/origin/1553 (bedbe76a65b4) (dirty)

35.20220223.dev.0
   Timestamp: 2022-02-23T14:06:22Z (12 days, 2:19:04 ago)
   Artifacts: ostree qemu
      Config: testing-devel (7413ad404988)

35.20220202.dev.0
   Timestamp: 2022-02-02T18:50:39Z (32 days, 21:34:47 ago)
   Artifacts: ostree qemu
      Config: testing-devel (6d7b9ad5c61a)
```

Author: sounds good, so should I just copy the contents of the kola.md file and add what I've done on the kola testiso aswell as the debug artifacts and just create a new PR?

Reviewer: You can reuse this PR if you like or create a new one...whatever is easiest for you.

# Kola Introduction Documentation

> This document goes through the basics of how kola tests work and how someone just beginning can get started

Reviewer 2: ```suggestion
This document goes through the basics of how kola tests work and how someone just beginning can get started.
```

### What is Kola?

Kola is a framework for testing software integration in CoreOS systems across multiple platforms. It is primarily designed to operate within the CoreOS Assembler for testing software that has landed in the OS image.

> [Additional documentation about Kola](https://github.com/coreos/coreos-assembler/tree/main/docs/kola)

### Steps to run a test from start to finish

 1. `cosa run` (Creates and brings you into the shell of a QEMU instance, running the provided version of FCOS or RHCOS)
 2. `cosa kola list` (Here we can see all of the tests that we can do)
 3. `cosa kola run <test-pattern>` (This will run a specific test and check to see if a test passes)
 4. `cosa kola run ext.config*` (This will run all the tests that start with `ext.config`)

### More information on tests

There are file logs in each of these locations: They will help you to debug the problem and will certainly give you hints along the way

1. `journal.txt`
2. `console.txt`
3. `ignition.json`
4. `gunzip journal-raw.txt.gz`

> ### Extended artifacts

Reviewer 2: the numbered list below has two `2.` entries

> 1. Extended artefacts need additional forms of testing (You can pass the ignition and the path to the artefact you want to test)

Reviewer 2: ```suggestion
1. Extended artifacts need additional forms of testing (You can pass the ignition and the path to the artifact you want to test)
```

Author: Thanks for all of these suggestions Dusty!

2. `cosa kola run -h` (this allows you to see the commands yourself and what syntax is needed)

> 2. `cosa buildextend-"name_of_artifact` (An example of building an artefact before testing it)

Reviewer 2: ```suggestion
2. `cosa buildextend-"name_of_artifact` (An example of building an artifact before testing it)
```

3. `kola run -p <platform>` Is the most generic way of testing extended artifacts, this is mostly useful for the cloud platforms
4. For running the likes of metal/metal4k artifacts there's not much difference than running `kola run` from the coreos-assembler
5. `cd builds/latest/` (This will show your latest build information)
6. In the case of the `testiso` command, you'll see that there is the `--qemu-native-4k` option passed to `kola testiso`.  This instructs the `testiso` test to attempt to install FCOS/RHCOS to a disk that uses 4k sector size.  If you don't include that option, the `testiso` command will attempt to install FCOS/RHCOS to a non 4k disk (512b sector size)
7. `kola testiso -S --scenarios pxe-install,pxe-offline-install --output-dir tmp/kola-metal`

> 8. `cosa kola testiso --qemu-native-4k` (This is an example of me testing the live ISO build for a 4k sectors disk, this tests all of the scenarios)

Reviewer 2: I would try to change the writing style to remove words like "me" from this document.

Example output:

```
kola -p qemu-unpriv --output-dir tmp/kola testiso -P --qemu-native-4k
Testing scenarios: [iso-offline-install iso-live-login iso-as-disk miniso-install miniso-install-nm]
Detected development build; disabling signature verification
Successfully tested scenario iso-offline-install for 35.20220217.dev.0 on uefi (metal4k)
Successfully tested scenario iso-live-login for 35.20220217.dev.0 on uefi (metal4k)
Successfully tested scenario iso-as-disk for 35.20220217.dev.0 on uefi (metal4k)
Successfully tested scenario miniso-install for 35.20220217.dev.0 on uefi (metal4k)
Successfully tested scenario miniso-install-nm for 35.20220217.dev.0 on uefi (metal4k with NM keyfile)
```

> > [This is a good starting point:](https://gitlab.cee.redhat.com/coreos/redhat-coreos/-/blob/master/upshift/osbuild.groovy#L318-371)

Reviewer 2: This is an internal link (not accessible to the public). We shouldn't use that here.

### Command explanation

`cosa kola run 'name_of_test'` This is how to run a single test, This is used to help debug specific tests in order to get a better understanding of the bug that's taking place. Once you run this command this test will be added to the tmp directory

`cosa kola basic` This will just run the basic tests

`cosa kola --parallel` This will by default run 3 tests in a row

> In order to see the logs for these tests you must enter the `tmp/kola/name_of_the_tests` and there you will find the logs (journal and console files, ignition used and so on)

Reviewer 2: ```suggestion
In order to see the logs for these tests you must enter the `tmp/kola/name_of_the_test` directory and there you will find the logs (journal and console files, ignition used and so on).
```

> `cosa run` This launches the build you created ( in this way you can access the image for troubleshooting ). Also check the option -c ( console).

Reviewer 2: Let's drop the extra spacing in the `()` 
```suggestion
`cosa run` This launches the build you created (in this way you can access the image for troubleshooting). Also check the option -c (console).
```

> `cosa run -i ignition_path` You can run it passing your igniton, or the igntion used in the the test that failed for troubleshooting reasons.

Reviewer 2: ```suggestion
`cosa run -i ignition_path` You can run it passing your Ignition, or the Ignition used in the the test that failed for troubleshooting reasons.
```
